export type ImageEntry = {
	id: string,
	assetId: string,
	name: string,
	thumbnailId: string,
	category: string,
	packId: string,
	defaultUnlocked: boolean,
}

local IMAGES: { ImageEntry } = {
	-- TEMP: Single puzzle for alt account testing. Uncomment all below when back on main account.
	{
		id = "landscapes_temp",
		assetId = "rbxassetid://135779913190012",
		name = "Landscape",
		thumbnailId = "rbxassetid://135779913190012",
		category = "Nature",
		packId = "landscapes",
		defaultUnlocked = true,
	},
--[[ TEMPORARILY DISABLED - all original puzzles below
	-- Landscapes pack
	{
		id = "landscapes_aurora",
		assetId = "rbxassetid://90596179099990",
		name = "Aurora",
		thumbnailId = "rbxassetid://90596179099990",
		category = "Nature",
		packId = "landscapes",
		defaultUnlocked = true,
	},
	{
		id = "landscapes_autumn_walk",
		assetId = "rbxassetid://103760254865759",
		name = "Autumn Walk",
		thumbnailId = "rbxassetid://103760254865759",
		category = "Nature",
		packId = "landscapes",
		defaultUnlocked = true,
	},
	{
		id = "landscapes_cityscape",
		assetId = "rbxassetid://133820685209329",
		name = "Cityscape",
		thumbnailId = "rbxassetid://133820685209329",
		category = "Nature",
		packId = "landscapes",
		defaultUnlocked = true,
	},
	{
		id = "landscapes_cove",
		assetId = "rbxassetid://77626131172182",
		name = "Cove",
		thumbnailId = "rbxassetid://77626131172182",
		category = "Nature",
		packId = "landscapes",
		defaultUnlocked = true,
	},
	{
		id = "landscapes_creek",
		assetId = "rbxassetid://112139569289265",
		name = "Creek",
		thumbnailId = "rbxassetid://112139569289265",
		category = "Nature",
		packId = "landscapes",
		defaultUnlocked = true,
	},
	{
		id = "landscapes_glacier_lake",
		assetId = "rbxassetid://74751261116950",
		name = "Glacier Lake",
		thumbnailId = "rbxassetid://74751261116950",
		category = "Nature",
		packId = "landscapes",
		defaultUnlocked = false,
	},
	{
		id = "landscapes_horses",
		assetId = "rbxassetid://114601300193274",
		name = "Horses",
		thumbnailId = "rbxassetid://114601300193274",
		category = "Nature",
		packId = "landscapes",
		defaultUnlocked = false,
	},
	{
		id = "landscapes_hot_air_balloons",
		assetId = "rbxassetid://120485494867714",
		name = "Hot Air Balloons",
		thumbnailId = "rbxassetid://120485494867714",
		category = "Nature",
		packId = "landscapes",
		defaultUnlocked = false,
	},
	{
		id = "landscapes_jungle_path",
		assetId = "rbxassetid://114447272005858",
		name = "Jungle Path",
		thumbnailId = "rbxassetid://114447272005858",
		category = "Nature",
		packId = "landscapes",
		defaultUnlocked = false,
	},
	{
		id = "landscapes_lavender",
		assetId = "rbxassetid://123780836947325",
		name = "Lavender",
		thumbnailId = "rbxassetid://123780836947325",
		category = "Nature",
		packId = "landscapes",
		defaultUnlocked = false,
	},
	{
		id = "landscapes_meadow",
		assetId = "rbxassetid://80170445988778",
		name = "Meadow",
		thumbnailId = "rbxassetid://80170445988778",
		category = "Nature",
		packId = "landscapes",
		defaultUnlocked = false,
	},
	{
		id = "landscapes_mountain_flowers",
		assetId = "rbxassetid://126521125192843",
		name = "Mountain Flowers",
		thumbnailId = "rbxassetid://126521125192843",
		category = "Nature",
		packId = "landscapes",
		defaultUnlocked = false,
	},
	{
		id = "landscapes_mountain_lake",
		assetId = "rbxassetid://98002443004026",
		name = "Mountain Lake",
		thumbnailId = "rbxassetid://98002443004026",
		category = "Nature",
		packId = "landscapes",
		defaultUnlocked = false,
	},
	{
		id = "landscapes_pasture",
		assetId = "rbxassetid://78440872581594",
		name = "Pasture",
		thumbnailId = "rbxassetid://78440872581594",
		category = "Nature",
		packId = "landscapes",
		defaultUnlocked = false,
	},
	{
		id = "landscapes_pink_sunset",
		assetId = "rbxassetid://112229553557989",
		name = "Pink Sunset",
		thumbnailId = "rbxassetid://112229553557989",
		category = "Nature",
		packId = "landscapes",
		defaultUnlocked = false,
	},
	{
		id = "landscapes_rainbow",
		assetId = "rbxassetid://116941694092172",
		name = "Rainbow",
		thumbnailId = "rbxassetid://116941694092172",
		category = "Nature",
		packId = "landscapes",
		defaultUnlocked = false,
	},
	{
		id = "landscapes_reflection",
		assetId = "rbxassetid://138159372483412",
		name = "Reflection",
		thumbnailId = "rbxassetid://138159372483412",
		category = "Nature",
		packId = "landscapes",
		defaultUnlocked = false,
	},
	{
		id = "landscapes_river",
		assetId = "rbxassetid://81674027685330",
		name = "River",
		thumbnailId = "rbxassetid://81674027685330",
		category = "Nature",
		packId = "landscapes",
		defaultUnlocked = false,
	},
	{
		id = "landscapes_sandstone",
		assetId = "rbxassetid://123943483360307",
		name = "Sandstone",
		thumbnailId = "rbxassetid://123943483360307",
		category = "Nature",
		packId = "landscapes",
		defaultUnlocked = false,
	},
	{
		id = "landscapes_shore",
		assetId = "rbxassetid://129335012351588",
		name = "Shore",
		thumbnailId = "rbxassetid://129335012351588",
		category = "Nature",
		packId = "landscapes",
		defaultUnlocked = false,
	},
	{
		id = "landscapes_sunset_mountains",
		assetId = "rbxassetid://139288168118821",
		name = "Sunset Mountains",
		thumbnailId = "rbxassetid://139288168118821",
		category = "Nature",
		packId = "landscapes",
		defaultUnlocked = false,
	},
	{
		id = "landscapes_sunset_tree",
		assetId = "rbxassetid://72095431407726",
		name = "Sunset Tree",
		thumbnailId = "rbxassetid://72095431407726",
		category = "Nature",
		packId = "landscapes",
		defaultUnlocked = false,
	},
	{
		id = "landscapes_tree",
		assetId = "rbxassetid://87324176486940",
		name = "Tree",
		thumbnailId = "rbxassetid://87324176486940",
		category = "Nature",
		packId = "landscapes",
		defaultUnlocked = false,
	},
	{
		id = "landscapes_waterfall",
		assetId = "rbxassetid://99052367636599",
		name = "Waterfall",
		thumbnailId = "rbxassetid://99052367636599",
		category = "Nature",
		packId = "landscapes",
		defaultUnlocked = false,
	},
	{
		id = "city_skyline",
		assetId = "rbxassetid://108408685486498",
		name = "City Skyline",
		thumbnailId = "rbxassetid://108408685486498",
		category = "Urban",
		packId = "urban",
		defaultUnlocked = false,
	},
	{
		id = "space_nebula",
		assetId = "rbxassetid://127494809663498",
		name = "Space Nebula",
		thumbnailId = "rbxassetid://127494809663498",
		category = "Space",
		packId = "space",
		defaultUnlocked = false,
	},
	{
		id = "abstract_art",
		assetId = "rbxassetid://139907825498498",
		name = "Abstract Art",
		thumbnailId = "rbxassetid://139907825498498",
		category = "Art",
		packId = "brain",
		defaultUnlocked = false,
	},
	{
		id = "seal_beach",
		assetId = "rbxassetid://80051853985966",
		name = "Seal Beach",
		thumbnailId = "rbxassetid://80051853985966",
		category = "Animals",
		packId = "seals",
		defaultUnlocked = true,
	},
	{
		id = "sushi_platter",
		assetId = "",
		name = "Sushi Platter",
		thumbnailId = "",
		category = "Food",
		packId = "food",
		defaultUnlocked = true,
	},
	{
		id = "tokyo_street",
		assetId = "",
		name = "Tokyo Street",
		thumbnailId = "",
		category = "Urban",
		packId = "urban",
		defaultUnlocked = false,
	},
	{
		id = "seal_portrait",
		assetId = "rbxassetid://111469769800178",
		name = "Seal Portrait",
		thumbnailId = "rbxassetid://111469769800178",
		category = "Animals",
		packId = "seals",
		defaultUnlocked = false,
	},
	{
		id = "mars_surface",
		assetId = "",
		name = "Mars Surface",
		thumbnailId = "",
		category = "Space",
		packId = "space",
		defaultUnlocked = false,
	},
	{
		id = "watercolor_flowers",
		assetId = "",
		name = "Watercolor Flowers",
		thumbnailId = "",
		category = "Art",
		packId = "art",
		defaultUnlocked = false,
	},
	{
		id = "pizza_slice",
		assetId = "",
		name = "Pizza Slice",
		thumbnailId = "",
		category = "Food",
		packId = "food",
		defaultUnlocked = false,
	},
	{
		id = "rainy_window",
		assetId = "",
		name = "Rainy Window",
		thumbnailId = "",
		category = "Urban",
		packId = "urban",
		defaultUnlocked = false,
	},
	{
		id = "seal_rocks",
		assetId = "rbxassetid://123106766909130",
		name = "Seal Rocks",
		thumbnailId = "rbxassetid://123106766909130",
		category = "Animals",
		packId = "seals",
		defaultUnlocked = false,
	},
	{
		id = "galaxy_spiral",
		assetId = "",
		name = "Galaxy Spiral",
		thumbnailId = "",
		category = "Space",
		packId = "space",
		defaultUnlocked = false,
	},
	{
		id = "oil_painting",
		assetId = "",
		name = "Oil Painting",
		thumbnailId = "",
		category = "Art",
		packId = "art",
		defaultUnlocked = false,
	},
	{
		id = "ice_cream_sundae",
		assetId = "",
		name = "Ice Cream Sundae",
		thumbnailId = "",
		category = "Food",
		packId = "food",
		defaultUnlocked = false,
	},
	{
		id = "london_bridge",
		assetId = "",
		name = "London Bridge",
		thumbnailId = "",
		category = "Urban",
		packId = "urban",
		defaultUnlocked = false,
	},
	{
		id = "seal_closeup",
		assetId = "rbxassetid://116484971166421",
		name = "Seal Closeup",
		thumbnailId = "rbxassetid://116484971166421",
		category = "Animals",
		packId = "seals",
		defaultUnlocked = false,
	},
	{
		id = "saturn_rings",
		assetId = "",
		name = "Saturn Rings",
		thumbnailId = "",
		category = "Space",
		packId = "space",
		defaultUnlocked = false,
	},
	{
		id = "pixel_art",
		assetId = "",
		name = "Pixel Art",
		thumbnailId = "",
		category = "Art",
		packId = "art",
		defaultUnlocked = false,
	},
	{
		id = "pancake_stack",
		assetId = "",
		name = "Pancake Stack",
		thumbnailId = "",
		category = "Food",
		packId = "food",
		defaultUnlocked = false,
	},
	{
		id = "neon_alley",
		assetId = "",
		name = "Neon Alley",
		thumbnailId = "",
		category = "Urban",
		packId = "urban",
		defaultUnlocked = false,
	},
	{
		id = "earth_from_space",
		assetId = "",
		name = "Earth from Space",
		thumbnailId = "",
		category = "Space",
		packId = "space",
		defaultUnlocked = false,
	},
	-- Animals pack
	{
		id = "animals_1",
		assetId = "rbxassetid://82752581676436",
		name = "Animals I",
		thumbnailId = "rbxassetid://82752581676436",
		category = "Animals",
		packId = "animals",
		defaultUnlocked = true,
	},
	{
		id = "animals_2",
		assetId = "rbxassetid://82674555914169",
		name = "Animals II",
		thumbnailId = "rbxassetid://82674555914169",
		category = "Animals",
		packId = "animals",
		defaultUnlocked = true,
	},
	{
		id = "animals_3",
		assetId = "rbxassetid://108487459204397",
		name = "Animals III",
		thumbnailId = "rbxassetid://108487459204397",
		category = "Animals",
		packId = "animals",
		defaultUnlocked = true,
	},
	{
		id = "animals_4",
		assetId = "rbxassetid://125411386158587",
		name = "Animals IV",
		thumbnailId = "rbxassetid://125411386158587",
		category = "Animals",
		packId = "animals",
		defaultUnlocked = true,
	},
	{
		id = "animals_5",
		assetId = "rbxassetid://75497675318594",
		name = "Animals V",
		thumbnailId = "rbxassetid://75497675318594",
		category = "Animals",
		packId = "animals",
		defaultUnlocked = true,
	},
	{
		id = "animals_6",
		assetId = "rbxassetid://140501783343799",
		name = "Animals VI",
		thumbnailId = "rbxassetid://140501783343799",
		category = "Animals",
		packId = "animals",
		defaultUnlocked = false,
	},
	{
		id = "animals_7",
		assetId = "rbxassetid://92997878407723",
		name = "Animals VII",
		thumbnailId = "rbxassetid://92997878407723",
		category = "Animals",
		packId = "animals",
		defaultUnlocked = false,
	},
	{
		id = "animals_8",
		assetId = "rbxassetid://77142293121697",
		name = "Animals VIII",
		thumbnailId = "rbxassetid://77142293121697",
		category = "Animals",
		packId = "animals",
		defaultUnlocked = false,
	},
	{
		id = "animals_9",
		assetId = "rbxassetid://76635061159135",
		name = "Animals IX",
		thumbnailId = "rbxassetid://76635061159135",
		category = "Animals",
		packId = "animals",
		defaultUnlocked = false,
	},
	{
		id = "animals_10",
		assetId = "rbxassetid://128805616573201",
		name = "Animals X",
		thumbnailId = "rbxassetid://128805616573201",
		category = "Animals",
		packId = "animals",
		defaultUnlocked = false,
	},
	{
		id = "animals_11",
		assetId = "rbxassetid://130719640967651",
		name = "Animals XI",
		thumbnailId = "rbxassetid://130719640967651",
		category = "Animals",
		packId = "animals",
		defaultUnlocked = false,
	},
	{
		id = "animals_12",
		assetId = "rbxassetid://90012945648193",
		name = "Animals XII",
		thumbnailId = "rbxassetid://90012945648193",
		category = "Animals",
		packId = "animals",
		defaultUnlocked = false,
	},
	{
		id = "animals_13",
		assetId = "rbxassetid://133438494534513",
		name = "Animals XIII",
		thumbnailId = "rbxassetid://133438494534513",
		category = "Animals",
		packId = "animals",
		defaultUnlocked = false,
	},
	{
		id = "animals_14",
		assetId = "rbxassetid://71975467933275",
		name = "Animals XIV",
		thumbnailId = "rbxassetid://71975467933275",
		category = "Animals",
		packId = "animals",
		defaultUnlocked = false,
	},
	{
		id = "animals_15",
		assetId = "rbxassetid://80122601407125",
		name = "Animals XV",
		thumbnailId = "rbxassetid://80122601407125",
		category = "Animals",
		packId = "animals",
		defaultUnlocked = false,
	},
	{
		id = "animals_16",
		assetId = "rbxassetid://117906952555000",
		name = "Animals XVI",
		thumbnailId = "rbxassetid://117906952555000",
		category = "Animals",
		packId = "animals",
		defaultUnlocked = false,
	},
	{
		id = "animals_17",
		assetId = "rbxassetid://138928055258789",
		name = "Animals XVII",
		thumbnailId = "rbxassetid://138928055258789",
		category = "Animals",
		packId = "animals",
		defaultUnlocked = false,
	},
	{
		id = "animals_18",
		assetId = "rbxassetid://93754926804642",
		name = "Animals XVIII",
		thumbnailId = "rbxassetid://93754926804642",
		category = "Animals",
		packId = "animals",
		defaultUnlocked = false,
	},
	{
		id = "animals_19",
		assetId = "rbxassetid://137430575844464",
		name = "Animals XIX",
		thumbnailId = "rbxassetid://137430575844464",
		category = "Animals",
		packId = "animals",
		defaultUnlocked = false,
	},
	{
		id = "animals_20",
		assetId = "rbxassetid://128797585634207",
		name = "Animals XX",
		thumbnailId = "rbxassetid://128797585634207",
		category = "Animals",
		packId = "animals",
		defaultUnlocked = false,
	},
	{
		id = "animals_21",
		assetId = "rbxassetid://97198032106272",
		name = "Animals XXI",
		thumbnailId = "rbxassetid://97198032106272",
		category = "Animals",
		packId = "animals",
		defaultUnlocked = false,
	},
	{
		id = "animals_22",
		assetId = "rbxassetid://77510841794728",
		name = "Animals XXII",
		thumbnailId = "rbxassetid://77510841794728",
		category = "Animals",
		packId = "animals",
		defaultUnlocked = false,
	},
	{
		id = "animals_23",
		assetId = "rbxassetid://95986053621048",
		name = "Animals XXIII",
		thumbnailId = "rbxassetid://95986053621048",
		category = "Animals",
		packId = "animals",
		defaultUnlocked = false,
	},
	{
		id = "animals_24",
		assetId = "rbxassetid://110889795855038",
		name = "Animals XXIV",
		thumbnailId = "rbxassetid://110889795855038",
		category = "Animals",
		packId = "animals",
		defaultUnlocked = false,
	},
	{
		id = "animals_25",
		assetId = "rbxassetid://123014916266883",
		name = "Animals XXV",
		thumbnailId = "rbxassetid://123014916266883",
		category = "Animals",
		packId = "animals",
		defaultUnlocked = false,
	},
	{
		id = "animals_26",
		assetId = "rbxassetid://90165745200176",
		name = "Animals XXVI",
		thumbnailId = "rbxassetid://90165745200176",
		category = "Animals",
		packId = "animals",
		defaultUnlocked = false,
	},
	{
		id = "animals_27",
		assetId = "rbxassetid://133172159092974",
		name = "Animals XXVII",
		thumbnailId = "rbxassetid://133172159092974",
		category = "Animals",
		packId = "animals",
		defaultUnlocked = false,
	},
	{
		id = "animals_28",
		assetId = "rbxassetid://88091630998974",
		name = "Animals XXVIII",
		thumbnailId = "rbxassetid://88091630998974",
		category = "Animals",
		packId = "animals",
		defaultUnlocked = false,
	},
	{
		id = "animals_29",
		assetId = "rbxassetid://86124198139051",
		name = "Animals XXIX",
		thumbnailId = "rbxassetid://86124198139051",
		category = "Animals",
		packId = "animals",
		defaultUnlocked = false,
	},
	{
		id = "animals_30",
		assetId = "rbxassetid://123985532750170",
		name = "Animals XXX",
		thumbnailId = "rbxassetid://123985532750170",
		category = "Animals",
		packId = "animals",
		defaultUnlocked = false,
	},
	{
		id = "animals_31",
		assetId = "rbxassetid://122373784566619",
		name = "Animals XXXI",
		thumbnailId = "rbxassetid://122373784566619",
		category = "Animals",
		packId = "animals",
		defaultUnlocked = false,
	},
	{
		id = "animals_32",
		assetId = "rbxassetid://88574503559157",
		name = "Animals XXXII",
		thumbnailId = "rbxassetid://88574503559157",
		category = "Animals",
		packId = "animals",
		defaultUnlocked = false,
	},
	{
		id = "animals_33",
		assetId = "rbxassetid://127628051937116",
		name = "Animals XXXIII",
		thumbnailId = "rbxassetid://127628051937116",
		category = "Animals",
		packId = "animals",
		defaultUnlocked = false,
	},
	{
		id = "animals_34",
		assetId = "rbxassetid://106415720277208",
		name = "Animals XXXIV",
		thumbnailId = "rbxassetid://106415720277208",
		category = "Animals",
		packId = "animals",
		defaultUnlocked = false,
	},
	{
		id = "animals_35",
		assetId = "rbxassetid://121366753602573",
		name = "Animals XXXV",
		thumbnailId = "rbxassetid://121366753602573",
		category = "Animals",
		packId = "animals",
		defaultUnlocked = false,
	},
	-- Brainrots pack
	{
		id = "brainrot_1",
		assetId = "rbxassetid://110880891769071",
		name = "Brainrot I",
		thumbnailId = "rbxassetid://110880891769071",
		category = "Brainrot",
		packId = "brainrots",
		defaultUnlocked = true,
	},
	{
		id = "brainrot_2",
		assetId = "rbxassetid://91420917331650",
		name = "Brainrot II",
		thumbnailId = "rbxassetid://91420917331650",
		category = "Brainrot",
		packId = "brainrots",
		defaultUnlocked = true,
	},
	{
		id = "brainrot_3",
		assetId = "rbxassetid://83674904010561",
		name = "Brainrot III",
		thumbnailId = "rbxassetid://83674904010561",
		category = "Brainrot",
		packId = "brainrots",
		defaultUnlocked = true,
	},
	{
		id = "brainrot_4",
		assetId = "rbxassetid://105903466794995",
		name = "Brainrot IV",
		thumbnailId = "rbxassetid://105903466794995",
		category = "Brainrot",
		packId = "brainrots",
		defaultUnlocked = true,
	},
	{
		id = "brainrot_5",
		assetId = "rbxassetid://101708869824729",
		name = "Brainrot V",
		thumbnailId = "rbxassetid://101708869824729",
		category = "Brainrot",
		packId = "brainrots",
		defaultUnlocked = true,
	},
	{
		id = "brainrot_6",
		assetId = "rbxassetid://110027847402788",
		name = "Brainrot VI",
		thumbnailId = "rbxassetid://110027847402788",
		category = "Brainrot",
		packId = "brainrots",
		defaultUnlocked = false,
	},
	{
		id = "brainrot_7",
		assetId = "rbxassetid://135199903001404",
		name = "Brainrot VII",
		thumbnailId = "rbxassetid://135199903001404",
		category = "Brainrot",
		packId = "brainrots",
		defaultUnlocked = false,
	},
	{
		id = "brainrot_8",
		assetId = "rbxassetid://98438189648779",
		name = "Brainrot VIII",
		thumbnailId = "rbxassetid://98438189648779",
		category = "Brainrot",
		packId = "brainrots",
		defaultUnlocked = false,
	},
	{
		id = "brainrot_9",
		assetId = "rbxassetid://120211358310330",
		name = "Brainrot IX",
		thumbnailId = "rbxassetid://120211358310330",
		category = "Brainrot",
		packId = "brainrots",
		defaultUnlocked = false,
	},
	{
		id = "brainrot_10",
		assetId = "rbxassetid://98460307302840",
		name = "Brainrot X",
		thumbnailId = "rbxassetid://98460307302840",
		category = "Brainrot",
		packId = "brainrots",
		defaultUnlocked = false,
	},
	{
		id = "brainrot_11",
		assetId = "rbxassetid://130844003147485",
		name = "Brainrot XI",
		thumbnailId = "rbxassetid://130844003147485",
		category = "Brainrot",
		packId = "brainrots",
		defaultUnlocked = false,
	},
	{
		id = "brainrot_12",
		assetId = "rbxassetid://102416844317530",
		name = "Brainrot XII",
		thumbnailId = "rbxassetid://102416844317530",
		category = "Brainrot",
		packId = "brainrots",
		defaultUnlocked = false,
	},
	{
		id = "brainrot_13",
		assetId = "rbxassetid://82230205515908",
		name = "Brainrot XIII",
		thumbnailId = "rbxassetid://82230205515908",
		category = "Brainrot",
		packId = "brainrots",
		defaultUnlocked = false,
	},
	{
		id = "brainrot_14",
		assetId = "rbxassetid://118414581182264",
		name = "Brainrot XIV",
		thumbnailId = "rbxassetid://118414581182264",
		category = "Brainrot",
		packId = "brainrots",
		defaultUnlocked = false,
	},
	{
		id = "brainrot_15",
		assetId = "rbxassetid://121219696309297",
		name = "Brainrot XV",
		thumbnailId = "rbxassetid://121219696309297",
		category = "Brainrot",
		packId = "brainrots",
		defaultUnlocked = false,
	},
	{
		id = "brainrot_16",
		assetId = "rbxassetid://99371082995926",
		name = "Brainrot XVI",
		thumbnailId = "rbxassetid://99371082995926",
		category = "Brainrot",
		packId = "brainrots",
		defaultUnlocked = false,
	},
	{
		id = "brainrot_17",
		assetId = "rbxassetid://117082741990201",
		name = "Brainrot XVII",
		thumbnailId = "rbxassetid://117082741990201",
		category = "Brainrot",
		packId = "brainrots",
		defaultUnlocked = false,
	},
	{
		id = "brainrot_18",
		assetId = "rbxassetid://109698504294521",
		name = "Brainrot XVIII",
		thumbnailId = "rbxassetid://109698504294521",
		category = "Brainrot",
		packId = "brainrots",
		defaultUnlocked = false,
	},
	{
		id = "brainrot_19",
		assetId = "rbxassetid://97585714647900",
		name = "Brainrot XIX",
		thumbnailId = "rbxassetid://97585714647900",
		category = "Brainrot",
		packId = "brainrots",
		defaultUnlocked = false,
	},
	{
		id = "brainrot_20",
		assetId = "rbxassetid://129072947626226",
		name = "Brainrot XX",
		thumbnailId = "rbxassetid://129072947626226",
		category = "Brainrot",
		packId = "brainrots",
		defaultUnlocked = false,
	},
	{
		id = "brainrot_21",
		assetId = "rbxassetid://102187332245607",
		name = "Brainrot XXI",
		thumbnailId = "rbxassetid://102187332245607",
		category = "Brainrot",
		packId = "brainrots",
		defaultUnlocked = false,
	},
	{
		id = "brainrot_22",
		assetId = "rbxassetid://80491825016999",
		name = "Brainrot XXII",
		thumbnailId = "rbxassetid://80491825016999",
		category = "Brainrot",
		packId = "brainrots",
		defaultUnlocked = false,
	},
	{
		id = "brainrot_23",
		assetId = "rbxassetid://131587649220691",
		name = "Brainrot XXIII",
		thumbnailId = "rbxassetid://131587649220691",
		category = "Brainrot",
		packId = "brainrots",
		defaultUnlocked = false,
	},
	{
		id = "brainrot_24",
		assetId = "rbxassetid://107622315373003",
		name = "Brainrot XXIV",
		thumbnailId = "rbxassetid://107622315373003",
		category = "Brainrot",
		packId = "brainrots",
		defaultUnlocked = false,
	},
	{
		id = "brainrot_25",
		assetId = "rbxassetid://82101432312198",
		name = "Brainrot XXV",
		thumbnailId = "rbxassetid://82101432312198",
		category = "Brainrot",
		packId = "brainrots",
		defaultUnlocked = false,
	},
--]] -- END TEMPORARILY DISABLED
}

local ImageCatalog = {}

-- Returns true if the image has valid (non-empty) asset IDs
local function hasValidAssets(entry: ImageEntry): boolean
	return entry.assetId ~= nil and entry.assetId ~= ""
end

function ImageCatalog.getAll(): { ImageEntry }
	local result = {}
	for _, entry in IMAGES do
		if hasValidAssets(entry) then
			table.insert(result, entry)
		end
	end
	return result
end

function ImageCatalog.getById(id: string): ImageEntry?
	for _, entry in IMAGES do
		if entry.id == id and hasValidAssets(entry) then
			return entry
		end
	end
	return nil
end

function ImageCatalog.getByPackId(packId: string): { ImageEntry }
	local result = {}
	for _, entry in IMAGES do
		if entry.packId == packId then
			table.insert(result, entry)
		end
	end
	return result
end

function ImageCatalog.getDefaultUnlocked(): { ImageEntry }
	local result = {}
	for _, entry in IMAGES do
		if entry.defaultUnlocked and hasValidAssets(entry) then
			table.insert(result, entry)
		end
	end
	return result
end

return ImageCatalog
